src/plugins/dist_matrixops/dist_diag.cpp

```
@@ -0,0 +1,511 @@
+                    }
+
+                    std::string tiling_type = "sym";
+                    if (valid(args[2]))
+                    {
+                        tiling_type = extract_string_value(
+                            std::move(args[2]), this_->name_, this_->codename_);
+                        if ((tiling_type != "sym" && tiling_type != "row") &&
+                            tiling_type != "column")
+                        {
+                            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                                "dist_diag::eval",
+                                this_->generate_error_message(
+                                    "invalid tiling_type. the tiling_type cane "
+                                    "one of these: `sym`, `row` or `column`"));
+                        }
+                    }
+
+                    std::uint32_t tile_idx = hpx::get_locality_id();
+                    if (valid(args[3]))
+                    {
+                        tile_idx = extract_scalar_nonneg_integer_value_strict(
+                            std::move(args[3]), this_->name_, this_->codename_);
+                    }
+                    std::uint32_t numtiles =
+                        hpx::get_num_localities(hpx::launch::sync);
+                    if (valid(args[4]))
+                    {
+                        numtiles = extract_scalar_positive_integer_value_strict(
+                            std::move(args[4]), this_->name_, this_->codename_);
+                    }
+                    if (tile_idx >= numtiles)
+                    {
+                        HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                            "dist_diag::eval",
+                            this_->generate_error_message(
+                                "invalid tile index. Tile indices start from 0 "
+                                "and should be smaller than number of tiles"));
+                    }
+
+
```

The user might supply `a` with any dimension rather than 1. Would you please throw an exception in the case that `a`'s dimension is not 1?

---

I am not sure I understand. Do you mean arr when mentioning a? And, why arr's dimension has to be 1? Or do you mean that it should be larger or equal to 1?

---

I mean if the user asks for the following (a 2-d array) what would be the result:
```
diag_d(annotate_d([[1,2,3],[4,5,6]],"name", list("tile",list("columns",0,3),list("rows",0,2))))
```

---

ahh got you.because we currently only consider 1d array as input.